src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/HTMLCSharpLanguageServerLogHubLoggerProvider.cs

```diff
@@ -42,14 +44,25 @@ internal HTMLCSharpLanguageServerLogHubLoggerProvider()
 
             _loggerFactory = loggerFactory;
 
+            _initializationSemaphore = new SemaphoreSlim(initialCount: 1, maxCount: 1);
+
             CreateMarkerFeedbackLoggerFile(feedbackLoggerProvider);
         }
 
         public async Task InitializeLoggerAsync()
         {
-            if (_loggerProvider is null)
+            await _initializationSemaphore.WaitAsync().ConfigureAwait(false);
```

Comment:
Ah nice catch. This should take a cancellation token though.

Also, to avoid blocking / hot path this you may want to add an extra `_loggerProvider is null` check above and not take the semaphore at all if you know it's not null